test/k8sT/Policies.go

```diff
@@ -808,3 +815,284 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
+
+	Expect(err).NotTo(HaveOccurred())
+
+	return pod
+}
+
+// testCanConnect creates and tests if a given pod can connect to a given
+// service on a specific destination port.
+func testCanConnect(k *helpers.Kubectl, ns, podName string, service *v1.Service, dPort int) {
+	pod := createNetworkClientPod(k, ns, podName, service, dPort)
+	defer func() {
+		By(fmt.Sprintf("Cleaning up the pod %s", podName))
+		err := k.CoreV1().Pods(ns).Delete(pod.Name, nil)
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been deleted", pod.Name))
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err := k.WaitForPodReady(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been ready", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err = k.WaitForPodFinish(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have finish successfully", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	success, err := k.WaitForPodSuccess(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), "Pod did not finish as expected.")
+	Expect(success).To(BeTrue(), "Unable to connect to service %s on port %d. (It should)", service.String(), dPort)
+}
+
+// testCannotConnect creates and tests if a given pod can not connect to a given
+// service on a specific destination port.
+func testCannotConnect(k *helpers.Kubectl, ns, podName string, service *v1.Service, dPort int) {
+	pod := createNetworkClientPod(k, ns, podName, service, dPort)
+	defer func() {
+		By(fmt.Sprintf("Cleaning up the pod %s", podName))
+		err := k.CoreV1().Pods(ns).Delete(pod.Name, nil)
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been deleted", pod.Name))
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err := k.WaitForPodReady(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been ready", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err = k.WaitForPodFinish(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have finish successfully", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	success, err := k.WaitForPodSuccess(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), "Pod did not finish as expected.")
+	Expect(success).To(BeFalse(), "Able to connect to service %s on port %d. (It shouldn't)", service.String(), dPort)
+}
+
+func cleanupNetworkPolicy(k *helpers.Kubectl, policy *networkingv1.NetworkPolicy) {
+	By("Cleaning up the policy.")
+	err := k.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, nil)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+var _ = Describe("K8sValidatedPolicyTestNamespaceSelector", func() {
```

Comment:
I would not create a new Testcase for this, I should add a new testcase in "K8sValidatedPolicyTest" like `"Same Policy in the different namespaces`